### PR TITLE
TimeoutExtender: Cap max extension time

### PR DESCRIPTION
**Pull request by TomFrost**
This PR also extends the timeout by _exactly_ the timeout instead of adding 5 seconds, due to SQS's ChangeTimeoutVisibility API call working additively instead of replacing the current timeout with the new one.

### CODE REVIEW

File: src/TimeoutExtender.js [hunk 1]
```diff
@@ -12,12 +12,19 @@
 const API_CALL_LEAD_MS = 5000
 
 /**
+ * The maximum age, in mulliseconds, that a message can reach before AWS will no longer accept VisibilityTimeout
```
Comment (stephen-ta) [inline src/TimeoutExtender.js]: `mulliseconds` typo

File: src/TimeoutExtender.js [hunk 2]
```diff
@@ -12,12 +12,19 @@
...
+const MAX_MESSAGE_AGE_MS = 43200000
+
+/**
  * Option defaults.
  * @type {Object}
  */
 const optDefaults = {
   visibilityTimeoutSecs: 30,
-  noExtensionsAfterSecs: 0
+  noExtensionsAfterSecs: MAX_MESSAGE_AGE_MS
```
Comment (stephen-ta) [inline src/TimeoutExtender.js]: Your variable dictates seconds, but you are providing it in milliseconds

File: src/TimeoutExtender.js [hunk 3]
```diff
@@ -52,7 +59,7 @@ class TimeoutExtender {
     this._squiss.on('handled', msg => this.deleteMessage(msg))
     this._squiss.on('message', msg => this.addMessage(msg))
     this._visTimeout = this._opts.visibilityTimeoutSecs * 1000
-    this._stopAfter = this._opts.noExtensionsAfterSecs * 1000
+    this._stopAfter = Math.min(this._opts.noExtensionsAfterSecs * 1000, MAX_MESSAGE_AGE_MS)
```
Comment (stephen-ta) [inline src/TimeoutExtender.js]: If this is the milliseconds time, you are extra `* 1000` then?

Comment (stephen-ta) [inline src/TimeoutExtender.js]: I guess either way, the `Math.min` will still end up with `MAX_MESSAGE_AGE_MS` so maybe you intended to allow this?!

File: src/TimeoutExtender.js (hunk 2 above)
Comment (TomFrost) [inline src/TimeoutExtender.js]: @%$^#)^%(@#)$^&()@

File: src/TimeoutExtender.js (hunk 3 above)
Comment (TomFrost) [inline src/TimeoutExtender.js]: Yeah, this line is right, it's the default that's wrong

**Pull request merged**